### PR TITLE
chore: parametrize services Dockerfile path

### PR DESCRIPTION
**Pull request by rogercoll**
# Changes

Define services paths and Dockerfiles using and environment variable so it can be easily overridden. With the added variables, a vendor can easily point to another Dockerfile using the `.env.override` file:

```
PAYMENT_SERVICE_DOCKERFILE=./src/paymentservice/Dockerfile.custom
```

That would help to make changes additive, thus minimizing merge conflicts. 
In addition, the demo can also benefit from it and add different Dockerfile for the services and reflect different Otel deployment strategies. 

The PR also aligns the `emailservice` Dockerfile with the other's services strategy by expecting to be built from the root's repository path.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

~~* [ ] `CHANGELOG.md` updated to document new feature additions~~
~~* [ ] Appropriate documentation updates in the [docs][]~~
~~* [ ] Appropriate Helm chart updates in the [helm-charts][]~~

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Comment by julianocosta89** [comment]
Hey @rogercoll 👋🏽 

This is interesting, and I like where this is going.
One note thought, would you be able to change the env vars to contain the whole path?

That would avoid the concatenation in the compose file and it would help when things do not go as expected.
WDYT?

**Comment by rogercoll** [comment]
>  One note thought, would you be able to change the env vars to contain the whole path?

Definitely, it will be even more helpful for vendors that want to completely move a service to a different directory. Something like https://github.com/open-telemetry/opentelemetry-demo/pull/1600/commits/8a40cd048891c88d14bd999ff0a984ea782aa524 ?

PS I will bring this PR/idea to today's SIG meeting

File: src/emailservice/Dockerfile [hunk 1]
```diff
@@ -8,15 +8,15 @@ FROM base as builder
 
 WORKDIR /tmp
 
-COPY Gemfile Gemfile.lock .
+COPY ./src/emailservice/Gemfile ./src/emailservice/Gemfile.lock ./
```
Comment (rogercoll) [inline src/emailservice/Dockerfile:11]: Use `Dockerfile` from root's demo directory instead (like the other services approach)

**Comment by julianocosta89** [comment]
@rogercoll, with 2 env vars to do the solution you mention we would need to update 2 places:

```
PAYMENT_SERVICE_DOCKERFILE=./src/paymentservice/Dockerfile.custom
```

My initial thought was having 1 env var containing the full path, do you see other use cases where having 2 would make more sense?

**Comment by puckpuck** [comment]
I would like to use a single variable for each service containing the path and name of the Dockerfile. If someone needs to update this, they only need to update a single spot to reflect either a new folder or a new Dockerfile name.

**Review by puckpuck — APPROVED** [review]
Thanks for this.

**Pull request merged**